nodes/boolean/compare.py

```diff
@@ -50,6 +68,9 @@ def getWantedDataType(self):
         if inputB is not None: return inputB.dataType
         return self.inputs[0].dataType
 
+    def assignAdvancedDataType(self, dataType):
```

Comment:
why did you implement this function? Instead of directly invoking the assignType function.